### PR TITLE
I18N: Collect stats on number of untranslated strings in message catalogues

### PR DESCRIPTION
**Pull request by joshhunt**
Quick CI stats addition to collect the number of untranslated strings in the grafana.json message catalogues.

Idea being we can monitor (and alert on) if the number of untranslated strings grows too high.

### CODE REVIEW

**Review by leventebalogh — APPROVED** [review]
Nice one, LGTM 👍

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: scripts/cli/reportI18nStats.mjs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/// @ts-check
+
+import { readdir, stat, readFile } from 'fs/promises';
+import path from 'path';
+
+const LOCALES_DIR = path.join('.', 'public', 'locales');
```
Comment (academo) [inline scripts/cli/reportI18nStats.mjs:6]: better if you start building this route from `__dirname` and `path.resolve` so you work with absolute paths

File: scripts/cli/reportI18nStats.mjs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+function eachMessage(value, callback) {
+  if (typeof value === 'object') {
+    for (const key in value) {
+      const element = value[key];
+      eachMessage(element, callback);
+    }
+  } else if (typeof value === 'string') {
+    callback(value);
+  } else {
+    throw new Error(`Unknown value ${value} in eachMessage`);
```
Comment (academo) [inline scripts/cli/reportI18nStats.mjs:75]: Is it intended to fail the script here? do you want to put some kind of try/catch where you call this function?

File: scripts/cli/reportI18nStats.mjs (hunk 1 above)
Comment (joshhunt) [reply scripts/cli/reportI18nStats.mjs]: nice!

**Pull request merged**